doc/man7/openssl-threads.pod

```diff
@@ -0,0 +1,98 @@
+=pod
+
+=head1 NAME
+
+openssl-threads - Overview of thread-safety in OpenSSL
+
+=head1 DESCRIPTION
+
+In this manpage, we use the term B<thread-safe> to indicate that an
+object can be used by multiple threads at the same time.
+
+OpenSSL can be built with or without threads support. The most important
+use of this support is so that OpenSSL itself can use a single consistent
+API, as shown in L<CRYPTO_THREAD_run_once(3)/EXAMPLES>.
+Multi-platform applications can also use this API.
+
+In particular, being configured for threads support does not imply that
+all OpenSSL objects are thread-safe.
+To repeat: I<most objects are not safe for simultaneous use>.
+Exceptions to this should be documented on the specific manual pages, and
+some general high-level guidance is given here.
+
+One major use of the OpenSSL thread API is to implement reference counting.
+Many object within OpenSSL are reference-counted, so resources are not
+released, until the last reference is removed.
+References are often increased automatically (such as when an B<X509>
+certificate object is added into an B<X509_STORE> trust store).
+There is often an I<object>_up_ref() API that can be used to increase the
+reference.
+Failure to match I<object>_up_ref() calls with the right number of
+I<object>_free() calls is a common source of memory leaks when a program
+exits.
+
+The various C<set1> and C<get1> functions associated with an object are
+used to set or get attributes of an object; they increase the refernece
+count.
```

Comment:
This seems quite confusing. They don't increase the ref count of the object. They *may* increase the ref count of the thing being set into or got from the object if the thing is a ref countable object. 

Also, typo: reference.

I find this whole discussion about ref counting useful, but misplaced on a page about threads. This is only very tangentially related to threads. Ref counting may be used as a mechanism to share a single object across multiple threads - but there are plenty of other scenarios where ref counting is useful.

Reply:
Fixed typo.  Added "increase the reference count if it's an object that has them."

It might be misplaced -- I disagree, reference counts and ownership are related to multi-threaded programming -- but I think it's important to explain somewhere. Perhaps a future edit will create an openssl-refcount page, but for now I think it should stay.